src/components/Pin.tsx

```diff
@@ -0,0 +1,73 @@
+import { Pressable, StyleSheet, Text, View, ViewStyle } from 'react-native'
+
+interface ComponentProps {
+  onPress: () => void,
+  rating: number,
+  style?: ViewStyle,
+}
+
+/**
+ * Pin Component -
+ *
+ * @param onPress the function that gets called when the pin is pressed
+ * @param rating the rating of the business that the pin is pointing to, it is also displayed on the pin
+ *               and used to generate the colour of the pin
+ * @param style the styling for the pressable so it can be moved and scaled if needed
+ * @returns a custom pin component
+ */
+const Pin = ({onPress, rating, style}: ComponentProps) => {
+
+  // if the rating is > 66 then the pin will be green, between 33 and 66 the pin will be yellow, and red if < 33
+  const colour = rating > 33 ? rating > 66 ? ['#1ea853', '#188441'] : ['#bed62e', '#a8bd29'] : ['#e2382d', '#842b18']
```

Comment:
It would be good to extract these to a common location so that they can be reused.